test/Sprites/ArtWriter.test.cpp

```diff
@@ -21,3 +23,40 @@ TEST(ArtWriter, BlankFilename)
 {
 	EXPECT_THROW(ArtFile::Write("", ArtFile()), std::runtime_error);
 }
+
+TEST(ArtWriter, scanLineByteWidth)
```

Comment:
The name of this test could be a little more clear about what is being tested. It not immediately obvious from looking at the code. In particular, I don't know why writing should throw.

I have noticed the `scanlineByteWidth` is 2 byte aligned, but not 4 byte aligned. It also seems kind of arbitrary in relation to the `width`. Not sure if that is related to the throwing though.

Additionally, is it necessarily to cleanup the output file if writing throws, or should the writing routine cleanup any partially written files? I suppose for debugging purposes we want those partial files around. Maybe something to think about for later once everything is tested and working.

Reply:
Since I am using the same filename for each test file, it seemed to make sense to delete them as I went. If I want to inspect the contents of a file, I typically just set a breakpoint before the delete call. It sort of keeps the clutter down when committing by not having all the produced test files laying around. All minor preference things. If you want to change this behavior to leaving the artifacts around, I don't mind.

Reply:
Err, my meaning was should the core library code (not the test code) delete the output file if writing fails part way through? I think that would make sense to do.

I suppose that would mean debugging requires setting a breakpoint in the error handler that cleans up the file, within the library, though that would likely trigger before the error was reported in any console messages. Though if the only console error reporting we do is the default stack trace when an exception bubbles up to the top and terminates the program, then nothing would really have changed debugging wise.

Reply:
Oh okay. Yeah, I agree that if an exception/failure occurs during writing, the file should be deleted. I'll tease this out and see if it is happening or not and fix if needed. We can worry about the test peace later if we notice a problem with getting the output for test purposes. So far it hasn't been a problem.